app/models/concerns/dirty_associations.rb

```diff
@@ -21,7 +21,8 @@ module ClassMethods
     def dirty_has_many_associations(*args)
       extension = Module.new do
         args.each do |association|
-          association_ids = association.to_s.singularize + '_ids'
+          association_ids = association.to_s
+          association_ids = association_ids.singularize + '_ids' unless association.to_s.end_with? '_ids'
```

Comment:
would you mind wrapping `'ids'` at the end with parentheses to visually glue it to `end_with?`, I think that's preferred way with statement modifiers sytnax